packages/ord-connect/src/components/SelectWalletModal/index.tsx

```diff
@@ -156,21 +159,51 @@ export function SelectWalletModal({
       onError(Wallet.XVERSE, err as Error);
       return false;
     }
-  };
+  }, [
+    closeModal,
+    disconnectWallet,
+    network,
+    onError,
+    updateAddress,
+    updateFormat,
+    updatePublicKey,
+    updateWallet,
+  ]);
 
   // Reconnect address change listener if there there is already a connected wallet
   useEffect(() => {
+    let isMounted = true;
+    let isConnectSuccessful = false;
+    const listener = () => onConnectUnisatWallet();
+
     if (wallet === Wallet.UNISAT && address && publicKey && format) {
       const connectToUnisatWalletOnLoad = async () => {
         const isUnisatExtensionReady = await waitForUnisatExtensionReady();
+        if (!isMounted) {
+          return;
+        }
         if (!isUnisatExtensionReady) {
           disconnectWallet();
           return;
         }
-        onConnectUnisatWallet(true);
+
+        isConnectSuccessful = await onConnectUnisatWallet({ readOnly: true });
+        if (!isMounted) {
```

Comment:
saw we have same check on line 182, need to check again?

Reply:
Yes, because these are async-await calls.

2 cases:
1. For line 182, 1st case, in strict mode, react will fire this twice, returning on the first one
2. For line 191, 2nd case, a user could navigate to some other page when this is unmounted then press Connect, and once line 190 resolves, it will add to the window event listener.